### ISSUE TITLE
Staging Review finding: Filtering previous questions results in change of context on input

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#104822]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Product Information
**Team:** Ask VA Integration
**Product:** Ask VA
**Feature:** Ask Va

## Findings details
**VA.gov Experience Standard - issue:** [User experiences predictable interactions with components and patterns.](https://design.va.gov/about/experience-standards/#user-can-identify-an-element-and-its-state)
**VA.gov Experience Standard - category:** [Usability](https://design.va.gov/about/experience-standards/#category-usability)
**Launch-blocking:** Yes
**Design System review:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

On [/ask-va/introduction](https://staging.va.gov/contact-us/ask-va/introduction) when authenticated and exploring questions you've asked before, the filters are applied and keyboard focus shifts to the "Showing X - Y..." message immediately after choosing an option in the select inputs.

The main point of reference here is [WCAG 3.2.2 - On input](https://www.w3.org/WAI/WCAG21/Understanding/on-input.html). Moving focus to the "Showing X - Y..." message when the filter action occurs is good, but having it occur on input may interrupt a user mid-task.

Three examples of why this matters:
- A sighted keyboard-only user knows that they want to filter by status and filter by category. After they choose their status, focus is sent to "Showing X - Y..." --- which means to filter by category, now they'll have to backtrack to the other input.
- A blind screen reader user filters by status and is sent to "Showing X - Y..." If they can't see that there's a filter by category input on screen, they've skipped over it entirely --- and may never realize that filtering by category is even an option.
- A blind screen reader user wants to filter by status. Select inputs can be used via keyboard two ways: opening them, arrowing through the options, and pressing enter to choose an option; or just arrowing up and down through the options. Just arrowing up and down is fewer keystrokes, and if you don't benefit from the visual of seeing options displayed in the open dropdown then just arrowing up and down is preferable. But arrowing up and down also changes the value of the input, triggering the change in focus --- which means while you're still exploring what your filtering options are, you may be abruptly be moved to "Showing X - Y..." on your first keystroke.

<details><summary>Link, screenshot or steps to recreate</summary>

!["Your questions" section with select inputs for "Filter by status" and "Filter by category"](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/working-with-vsp/vsp-collaboration-cycle/staging-review-images/62304-Brian-3.png?raw=true)

</details>

## Recommended action

The filter action should occur when a button is pressed, not on input. This ensures that the action happens when the user wants it to happen, and not when they're still examining their options.

See also the recently-published candidate [search results template](https://design.va.gov/templates/search-results) in the design system.

## References

- **Accessibility Defect Severity:** [1: Critical. Must be fixed before launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-1)
- **WCAG Success Criteria:** [WCAG 3.2.2 On Input - Level A](https://www.w3.org/TR/WCAG21/#on-input)
- **Modality:** Keyboard
- **Design System Component:** N/A
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A

---

### Next Steps for the VFS Team

_How did this touchpoint go? [Give feedback on the Collaboration Cycle](https://dj540s05.optimalworkshop.com/questions/f54355271969f160ef41182ba9ceb61c) at any time._
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [x] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by arielma-tw** [comment]
Issue has been resolved in https://github.com/department-of-veterans-affairs/ask-va/issues/1658
